src/runtime/event-type-mapping.ts

```diff
@@ -225,49 +229,53 @@ export const ContributionSpecificAnalyticsEventToGoogleAnalyticsEvent = {
 
 /**
  * Converts a propensity event enum into an analytics event enum.
- * @param {!Event|string} propensityEvent
- * @returns {!AnalyticsEvent}
  */
-export function publisherEventToAnalyticsEvent(propensityEvent) {
+export function publisherEventToAnalyticsEvent(
+  propensityEvent: Event
+): AnalyticsEvent {
   return PublisherEventToAnalyticsEvent[propensityEvent];
 }
 
 /**
  * Converts an analytics event enum into a propensity event enum.
- * @param {?AnalyticsEvent} analyticsEvent
- * @returns {?Event}
  */
-export function analyticsEventToPublisherEvent(analyticsEvent) {
-  return AnalyticsEventToPublisherEvent[analyticsEvent];
+export function analyticsEventToPublisherEvent(
+  analyticsEvent: AnalyticsEvent | null
+): Event | null {
+  return (
+    (analyticsEvent && AnalyticsEventToPublisherEvent[analyticsEvent]) || null
+  );
 }
 
 /**
  * Converts a publisher entitlement event enum into an array analytics events.
- * @param {!ShowcaseEvent} event
- * @returns {!Array<AnalyticsEvent>}
  */
-export function showcaseEventToAnalyticsEvents(event) {
+export function showcaseEventToAnalyticsEvents(
+  event: ShowcaseEvent
+): Array<AnalyticsEvent> {
   return ShowcaseEvents[event] || [];
 }
 
-export function analyticsEventToEntitlementResult(event) {
+export function analyticsEventToEntitlementResult(event: AnalyticsEvent) {
   return AnalyticsEventToEntitlementResult[event];
 }
 
 /**
  * Converts an analytics event enum into a Google Analytics event object.
- * @param {?AnalyticsEvent} event
- * @param {string} subscriptionFlow
- * @returns {?Object}
  */
-export function analyticsEventToGoogleAnalyticsEvent(event, subscriptionFlow) {
-  let gaEvent = null;
-  if (subscriptionFlow) {
-    if (subscriptionFlow == SubscriptionFlows.SUBSCRIBE) {
-      gaEvent = SubscriptionSpecificAnalyticsEventToGoogleAnalyticsEvent[event];
-    } else if (subscriptionFlow == SubscriptionFlows.CONTRIBUTE) {
-      gaEvent = ContributionSpecificAnalyticsEventToGoogleAnalyticsEvent[event];
-    }
+export function analyticsEventToGoogleAnalyticsEvent(
+  event: AnalyticsEvent | null,
+  subscriptionFlow: string
+): GoogleAnalyticsEvent | void {
+  if (!event) {
+    return;
```

Comment:
I think the original code would return null in this case, can we preserve that behavior? It's unusual for code to conditionally return void.

Reply:
In this case, I think returning `null` makes the code more complicated (adds branches) and less readable.
![image](https://user-images.githubusercontent.com/1216762/228615532-e6b4fd37-7e37-496a-bbf5-6baeae94ac63.png)

I think returning `void` works well here. The app only calls the method in one place, and checks whether the result is falsy.
![image](https://user-images.githubusercontent.com/1216762/228615758-69fbce0b-7876-4913-8a81-94f21ceb4a9d.png)

Reply:
I think the JS could return `undefined`/`void` sometimes too. Ex: if `subscriptionFlow` was defined and `event` was `null`